### PR TITLE
[darwin] Fix a variety of small bugs in system_controls

### PR DESCRIPTION
**Pull request by jkeljo**
* Wasn't displaying quad values
* Wasn't displaying some numeric values that are specified in rare ways
* Technically incorrect lookup of type name
* Technically incorrect passing of a buffer size parameter

### CODE REVIEW

File: osquery/tables/system/darwin/sysctl_utils.cpp [hunk 1]
```diff
@@ -76,8 +77,35 @@ void genControlInfo(int* oid,
...
     oid_type = ((size_t)response[0] & CTLTYPE);
+    if ((oid_type == 0 || oid_type == CTLTYPE_INT) && response_size > 4) {
+      // For whatever reason, macOS defines fewer CTLTYPE's than BSD, and
+      // sometimes uses the format character instead of (or in addition to)
+      // the CTLTYPE to specify the type. Here we detect a few such cases and
+      // map them to CTLTYPE's.
+      // TODO: Both CTLTYPE_INT and CTLTYPE_QUAD can be specified as unsigned
+      // using a similar method.
+      char type_char = response[4];
+      switch (type_char) {
```
Comment (obelisk) [inline osquery/tables/system/darwin/sysctl_utils.cpp]: Should this have a default case?

File: osquery/tables/system/darwin/sysctl_utils.cpp [hunk 2]
```diff
@@ -76,8 +77,35 @@ void genControlInfo(int* oid,
...
+        case 'S':
+          oid_type = CTLTYPE_STRUCT;
+          break;
+        case 'Q':
+          oid_type = CTLTYPE_QUAD;
+          break;
+      }
+    }
     if (oid_type < kControlTypes.size()) {
-      r["type"] = kControlTypes[((int)response[0])];
```
Comment (obelisk) [inline osquery/tables/system/darwin/sysctl_utils.cpp]: 👍

Thanks for also clearing up the c-style cast.

File: osquery/tables/system/darwin/sysctl_utils.cpp (hunk 1 above)
Comment (jkeljo) [reply osquery/tables/system/darwin/sysctl_utils.cpp]: The default is just to leave it as what it was. I can add a comment to that effect.

**Comment by theopolis** [comment]
ok to test

**Comment by osqueryer** [comment]
:no_good: The commit 722ade4443d3a8dbf1f6adb7883361191545f75a (Job results: [3435](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3435/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 722ade4443d3a8dbf1f6adb7883361191545f75a (Job results: [3843](https://jenkins.osquery.io/job/osqueryPullRequestWindows/3843/)) failed one or more tests (**Windows**).

**Comment by theopolis** [comment]
`make format_master` wants
```cpp
diff --git a/osquery/tables/system/darwin/sysctl_utils.cpp b/osquery/tables/system/darwin/sysctl_utils.cpp
index 8bd76dfb..e63ae8d4 100644
--- a/osquery/tables/system/darwin/sysctl_utils.cpp
+++ b/osquery/tables/system/darwin/sysctl_utils.cpp
@@ -86,23 +86,23 @@ void genControlInfo(int* oid,
       // using a similar method.
       char type_char = response[4];
       switch (type_char) {
-        case 'I':
-          oid_type = CTLTYPE_INT;
-          break;
-        case 'L':
-          if (sizeof(long) == sizeof(long long)) {
-            oid_type = CTLTYPE_QUAD;
-          } else if (sizeof(long) == sizeof(int)) {
-            oid_type = CTLTYPE_INT;
-          }
-          break;
-        case 'S':
-          oid_type = CTLTYPE_STRUCT;
-          break;
-        case 'Q':
+      case 'I':
+        oid_type = CTLTYPE_INT;
+        break;
+      case 'L':
+        if (sizeof(long) == sizeof(long long)) {
           oid_type = CTLTYPE_QUAD;
-          break;
-          // Otherwise leave the type as it was; we have no additional knowledge
+        } else if (sizeof(long) == sizeof(int)) {
+          oid_type = CTLTYPE_INT;
+        }
+        break;
+      case 'S':
+        oid_type = CTLTYPE_STRUCT;
+        break;
+      case 'Q':
+        oid_type = CTLTYPE_QUAD;
+        break;
+        // Otherwise leave the type as it was; we have no additional knowledge
       }
     }
     if (oid_type < kControlTypes.size()) {
```

**Comment by jkeljo** [comment]
Rebased; fixed format

**Pull request merged**